test/react-web/client/graphql/queries.test.tsx

```diff
@@ -2128,4 +2128,184 @@ describe('queries', () => {
+    const queryObservable1 = (client as any).queryManager.observableQueries['1'].observableQuery;
+
+    wrapper1.unmount();
+
+    expect(Object.keys((client as any).queryManager.observableQueries)).toEqual(['1']);
+
+    const wrapper2 = renderer.create(
+      <ApolloProvider client={client}>
+        <Container/>
+      </ApolloProvider>
+    );
+
+    expect(Object.keys((client as any).queryManager.observableQueries)).toEqual(['1']);
+    const queryObservable2 = (client as any).queryManager.observableQueries['1'].observableQuery;
+
+    expect(queryObservable1).toBe(queryObservable2);
+
+    wrapper2.unmount();
+
+    expect(Object.keys((client as any).queryManager.observableQueries)).toEqual(['1']);
+  });
+
+  it('will recycle `ObservableQuery`s when re-rendering a portion of the tree', done => {
+    const query = gql`query people { allPeople(first: 1) { people { name } } }`;
+    const data = { allPeople: { people: [ { name: 'Luke Skywalker' } ] } };
+    const networkInterface = mockNetworkInterface(
+      { request: { query }, result: { data } },
+      { request: { query }, result: { data } },
+    );
+    const client = new ApolloClient({ networkInterface, addTypename: false });
+    let remount: any;
+
+    class Remounter extends React.Component<any, any> {
+      state = {
+        showChildren: true,
+      };
+
+      componentDidMount () {
+        remount = () => {
+          this.setState({ showChildren: false }, () => {
+            setTimeout(() => {
+              this.setState({ showChildren: true });
+            }, 5);
+          });
+        }
+      }
+
+      render () {
+        return this.state.showChildren ? this.props.children : null;
+      }
+    }
+
+    @graphql(query)
+    class Container extends React.Component<any, any> {
+      render () {
+        return null;
+      }
+    }
+
+    const wrapper = renderer.create(
+      <ApolloProvider client={client}>
+        <Remounter>
+          <Container/>
+        </Remounter>
+      </ApolloProvider>
+    );
+
+    expect(Object.keys((client as any).queryManager.observableQueries)).toEqual(['1']);
+    const queryObservable1 = (client as any).queryManager.observableQueries['1'].observableQuery;
+
+    remount();
+
+    setTimeout(() => {
+      expect(Object.keys((client as any).queryManager.observableQueries)).toEqual(['1']);
+      const queryObservable2 = (client as any).queryManager.observableQueries['1'].observableQuery;
+      expect(queryObservable1).toBe(queryObservable2);
+
+      remount();
+
+      setTimeout(() => {
+        expect(Object.keys((client as any).queryManager.observableQueries)).toEqual(['1']);
+        const queryObservable3 = (client as any).queryManager.observableQueries['1'].observableQuery;
+        expect(queryObservable1).toBe(queryObservable3);
+
+        wrapper.unmount();
+        done();
+      }, 10);
+    }, 10);
+  });
+
+  it('will not recycle parallel GraphQL container `ObservableQuery`s', done => {
```

Comment:
Yep. It shows that if you render two components with the same query you get two observable queries, but if you unmount and remount one of the components it will recycle that observable query so that in total you only ever have two observable queries.